driver/driver.go

```diff
@@ -0,0 +1,102 @@
+package driver
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"time"
+
+	"github.com/brimsec/zq/proc"
+	"github.com/brimsec/zq/zbuf"
+	"github.com/brimsec/zq/zqd/api"
+)
+
+type Driver interface {
+	Warn(msg string) error
+	Write(channelID int, batch zbuf.Batch) error
+	ChannelEnd(channelID int, stats api.ScannerStats) error
```

Comment:
I don't think adding the `api.ScannerStats` is necessary here. They aren't/won't be specific to that channel anyways.

Reply:
This is just a transcription of existing search behavior. I agree with your point (and wondered why this is this way) but this PR isn't for changing the search protocol... we should change this in a follow-up. (And in general, stats/warnings/feedback is going to get a bunch of updates with the forthcoming work on #528).

Reply:
Filed https://github.com/brimsec/zq/issues/555 to track this.